prometheus/collectors/go_collector_latest_test.go

```diff
@@ -40,9 +44,331 @@ func TestGoCollectorMarshalling(t *testing.T) {
 		t.Errorf("json marshalling shoud not fail, %v", err)
 	}
 }
+func TestWithGoCollectorMemStatsMetricsDisabled(t *testing.T) {
+	expected := []string{
+		"go_gc_duration_seconds",
+		"go_goroutines",
+		"go_info",
+		"go_memstats_last_gc_time_seconds",
+		"go_threads",
+	}
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(NewGoCollector(
+		WithGoCollectorMemStatsMetricsDisabled(),
+	))
+	result, err := reg.Gather()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	got := []string{}
+	for _, r := range result {
+		got = append(got, r.GetName())
+	}
+
+	if !reflect.DeepEqual(got, expected) {
+		t.Errorf("got %v, want %v", got, expected)
+	}
+}
+
+func TestGoCollectorAllowList(t *testing.T) {
```

Comment:
Same here: https://github.com/prometheus/client_golang/blob/b94d7e25292465ac578c2e885e6d61e03a81ec4b/prometheus/go_collector_latest_test.go#L77

Reply:
Let's have these as examples for visual people like me.